impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java

```diff
@@ -332,7 +332,9 @@ public ConfigHolder generateConfigForSpecificNode(KafkaListenerSource kafkaListe
         putConfig(nodeConfiguration, "listener.security.protocol.map", formatListeners(protocolMap, ":"));
         putConfig(nodeConfiguration, "listeners", formatListeners(listeners, "://"));
         putConfig(nodeConfiguration, "early.start.listeners", String.join(",", earlyStart));
-        putConfig(nodeConfiguration, "advertised.listeners", formatListeners(advertisedListeners, "://"));
+        if (!advertisedListeners.isEmpty()) {
```

Comment:
It turns out 3.8.0 in the KRaft controller case is sensitive to this parameter being present, even though it is empty.

Reply:
😱

Reply:
Test added.